manifests/config/global/jenkins_config.pp

```diff
@@ -0,0 +1,24 @@
+# This private class is called from jenkins::config to manage Jenkins' config.xml
+#
+class jenkins::config::global::jenkins_config(
+  $localstatedir           = $::jenkins::localstatedir,
+  $user                    = $::jenkins::user,
+  $group                   = $::jenkins::group,
+  $jenkins_config_template = $::jenkins::jenkins_config_template,
+) {
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $config_file = "${localstatedir}/config.xml"
+
+  file { $config_file:
+    ensure  => present,
+    content => template($jenkins_config_template),
```

Comment:
I think it would be better to pass in the content string rather than a template name.  That doesn't force the choice between ERB, EPP, hiera lookup, etc. on the end-user.